feature/amazon/src/main/java/com/revenuecat/purchases/amazon/ProxyAmazonBillingActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.revenuecat.purchases.amazon
+
+import android.app.Activity
+import android.content.IntentFilter
+import android.os.Bundle
+import android.os.ResultReceiver
+
+class ProxyAmazonBillingActivity : Activity() {
+
+    internal var broadcastReceiver: ProxyAmazonBillingActivityBroadcastReceiver? = null
+    private val filter = IntentFilter("com.revenuecat.purchases").apply {
+        addAction("purchase_finished")
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        // Applying theme programmatically because when applying via AndroidManifest, theme is not being
+        // applied correctly.
+        // What happens is that applying @android:style/Theme.Translucent.NoTitleBar in the manifest works
+        // but applying a theme that has that theme as parent, the Activity is not translucent
+        setTheme(R.style.ProxyAmazonBillingActivityTheme)
+        super.onCreate(savedInstanceState)
+        broadcastReceiver = ProxyAmazonBillingActivityBroadcastReceiver(this)
+        applicationContext.registerReceiver(broadcastReceiver, filter)
+
+        if (savedInstanceState == null) {
+            val sku = intent.getStringExtra("sku")
+            val resultReceiver = intent.getParcelableExtra<ResultReceiver>("result_receiver")
+            val purchasingServiceProvider =
+                intent.getParcelableExtra<PurchasingServiceProvider>("service_provider")
```

Comment:
Another idea could be to create a `ProxyAmazonBillingActivityArgs` class that contains all these 3 and make it parcelable, so there is only one argument. That way is slightly safer since the arguments definition is also centralized